### ISSUE TITLE
Property from missing in TestContext

### ISSUE THREAD

**Issue by SalahAdDin** [jquense/yup#1631]
**Describe the bug**
I am working in a nested schema validation and i found [this issue](https://github.com/jquense/yup/issues/735) pretty useful and it leads us to [this other issue](https://github.com/jquense/yup/issues/398).

Inside of, i found the this [issue](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/49512), who is pretty related to my current bug:
```
Unsafe assignment of an `any` value.
Unsafe member access [1] on an `any` value.
Property 'from' does not exist on type 'TestContext<AnyObject>'
```
The problem is **form** clearly does exist, cause I'm getting it, but the property does not exist in the type.

In the aforementioned issue there is a workaround for this bug, but I didn't work for me.

It is noteworthy that that issue has two years, why does this issue still persist?

**To Reproduce**
The following code can give an idea about the problem with typing in nested schemas.
```
const schema = object({
  firstName: string()
    .required("Don't forget to fill your First Name.")
    .min(5, 'Minimum 10 characters.')
    .max(15, 'Maximum 50 characters.'),
  lastName: string()
    .required("Don't forget to fill your Last Name")
    .min(5, 'Minimum 25 characters.')
    .max(15, 'Maximum 300 characters.'),
  birthDate: date().required("Don't forget to fill your Birth Date."),
  country: string().required("Don't forget to select a Country."),
  allowance: number()
    .required("Don't forget to fill your Hollydays' Allowance.")
    .when('country', {
      is: 'Spain',
      then: number().min(30),
      otherwise: number().min(0),
    })
    .when('country', {
      is: 'Brazil',
      then: number().max(40),
    }),
  extraFields: object({
    maritalStatus: string().test(
      'required only if country  Spain or Ghana',
      "Don't forget to select your Marital Status.",
      (value, context) => {
        const { country } = context.from[1].value;

        return (country === 'Spain' || country === 'Ghana') && value !== null;
      },
    ),
  }),
}).required();
```

**Expected behavior**
It should not give any type error.

**Platform (please complete the following information):**
 - Version `0.32.11`

**Additional context**
This is a problem about types definition.

**Comment by asgeo1** [comment]
I'm having the same issue, interested to see if there is a solution

**Comment by SalahAdDin** [comment]
> I'm having the same issue, interested to see if there is a solution

I solved it by:
```
interface TestContextExtended {
  from: {
    value: IUser;
  }[];
}
``` 
[and](https://github.com/SalahAdDin/tests/blob/main/challenge-omnipresent/src/presentation/widgets/form/validation.schema.ts#L39):
```
const { from } = context as TestContext & TestContextExtended;
        const { country } = from[1].value;

        const spainOrGhana = country === 'Spain' || country === 'Ghana';
        const notNullWhenSpainOrGhana = spainOrGhana && value !== null;
        const neitherSpainNorGhana = country !== 'Spain' && country !== 'Ghana';

        return notNullWhenSpainOrGhana || neitherSpainNorGhana;
```

The same way the original comment did.

**Comment by kaguya3222** [comment]
@jquense Why is this closed? It's solved by a workaround, but I think context type should have `from` property from the box

**Comment by kaguya3222** [comment]
@jquense As I understand, fix of this issue will be present in the 1.0.0 release?

**Comment by SalahAdDin** [comment]
> @jquense As I understand, fix of this issue will be present in the 1.0.0 release?

is it true?

**Comment by kaguya3222** [comment]
> > @jquense As I understand, fix of this issue will be present in the 1.0.0 release?
> 
> is it true?

You can check this [file](https://github.com/jquense/yup/blob/master/src/util/createValidation.ts) in `main` branch. `from` property in TestContext is present, but last published version of package is `1.0.0-beta.7`. Changelog is [here](https://github.com/jquense/yup/blob/master/CHANGELOG.md).

**Comment by SalahAdDin** [comment]
> > > @jquense As I understand, fix of this issue will be present in the 1.0.0 release?
> > 
> > 
> > is it true?
> 
> You can check this [file](https://github.com/jquense/yup/blob/master/src/util/createValidation.ts) in the `main` branch. `from` property in TestContext is present, but the last published version of the package is `1.0.0-beta.7`. Changelog is [here](https://github.com/jquense/yup/blob/master/CHANGELOG.md).

I will check it, thank you.

**Comment by birgersp** [comment]
> `from` property in TestContext is present, but last published version of package is `1.0.0-beta.7`.

I couldn't wrap my head around that sentence, so just to translate for others:

**Yes**, the `from` property is typed in version 1.0.0-beta.7.
